apps/mull-api/src/app/user/user.service.ts

```diff
@@ -19,16 +21,23 @@ export class UserService {
     return this.userRepository.findOne(id);
   }
 
+  findByEmail(email: string): Promise<User[]> {
+    return this.userRepository.find({ where: { email } });
+  }
+
   async findAllFriends(id: number): Promise<User[]> {
     const { friends } = await this.userRepository.findOne(id, { relations: ['friends'] });
     return friends;
   }
 
   async create(userInput: CreateUserInput): Promise<User> {
-    const { dob, name, email, password } = userInput;
-    const newUser = this.userRepository.create({ name, dob, email, password });
-    await this.userRepository.save(newUser);
-    return newUser;
+    if (userInput.type === UserType.LOCAL) {
+      const salt = await genSalt(10);
+      const hashed = await hash(userInput.password, salt);
+      userInput.password = hashed;
+    }
+    const newUser = this.userRepository.create(userInput);
+    return await this.userRepository.save(newUser);
```

Comment:
I haven't tested but can you check if we can do this in one line like how we usually do?
```suggestion
    return await this.userRepository.save(userInput);
```

Reply:
I get this error when I try to write it that way. 
![image](https://user-images.githubusercontent.com/36024638/99921655-cc3d9980-2cf9-11eb-8c1b-fd6223768f70.png)
I also remember that I wrote it that way since I ran into the same problem.

Reply:
I was able to do it in one like so. Tested it and it works.
```suggestion
    return await this.userRepository.save({...userInput});
```